File: find-missing-and-repeated-values.py
class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        dict = {}
        n = len(grid)
        m = n*n

        for i in range(1,m+1):
            dict[i] = 0
        
        for i in range(n):
            for j in range(n):
                dict[grid[i][j]]+=1
        ans = [0,0]
        for i in range(1,m+1):
            if dict[i]==2:
                ans[0]=i
            elif dict[i]==0:
                ans[1]=i
        
        return ans
        

File: find-missing-and-repeated-values.cpp
class Solution {
public:
    vector<int> findMissingAndRepeatedValues(vector<vector<int>>& grid) {
        vector<int>ans(2,0);
        unordered_map<int,int>mp;
        int n = grid.size();
        int m = n*n;
        for(int i=1; i<=m; i++){
            mp[i] = 0;
        }
        for(int i=0; i<n; i++){
            for(int j=0; j<n; j++){
                mp[grid[i][j]]++;
            }
        }
        for(int i=1; i<=m; i++){
            if(mp[i]==2){
                ans[0]=i;
            }
            else if(mp[i]==0){
                ans[1]=i;
            }
        }
        return ans;
    }
};